File: 2015/12.jl
using JSON

function counter(obj, c, part)

    if typeof(obj) == Dict{String, Any}
        if part == 2 && "red" in values(obj)
            return c
        end
        for (k, v) in obj
            c = counter(v, c, part)
        end
    elseif typeof(obj) == Vector{Any}
        for i in obj
            c = counter(i, c, part)
        end 
    elseif typeof(obj) == String
        return c
    elseif typeof(obj) == Int64 
        c += obj
    end 

    return c 
end

function readFile(part)
    abacus = JSON.parsefile("instructions12a.txt")

    tot = counter(abacus, 0, part)

    if part == 1
        println("🎅 ", tot)
    else
        println("🎄 ", tot)
    end
end

readFile(1)
readFile(2)

File: 2015/12.py
import time, json

def counter(obj, part=1, to_ignore='', c=0):
    if isinstance(obj, dict):
        if part == 2 and to_ignore in obj.values():
            return 0
        for k, v in obj.items():
            c += counter(v, part, to_ignore)
        return c
    elif isinstance(obj, list):
        for i in obj:
            c += counter(i, part, to_ignore)
        return c
    elif isinstance(obj, str):
        return 0
    elif isinstance(obj, int):
        return obj
    return c

def loadInput():
    with open("instructions12a.txt", "r") as f:
        data = json.load(f)
    return data

def part1(inputs):
    c = counter(inputs)
    print (f"🎄 Part 1: {c}")

def part2(inputs):
    c = counter(inputs, to_ignore='red', part=2)
    print (f"🎄🎅 Part 2: {c}")
    
if __name__ == "__main__":
    title = "Day 12: JSAbacusFramework.io"
    sub = "-"*(len(title)+2)

    print()
    print(f" {title} ")
    print(sub)
    
    inputs = loadInput()
    
    t0 = time.time()
    part1(inputs)
    print ("Time: {:.5f}".format(time.time()-t0))
    
    t0 = time.time()
    part2(inputs)
    print ("Time: {:.5f}".format(time.time()-t0))

